code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -156,7 +156,16 @@
 		if (src.floorrunning)
 			src.end_floorrunning(TRUE)
 		// move controller out
-		controller.set_loc(get_turf(src))
+		if (src.z == Z_LEVEL_STATION)
+			controller.set_loc(get_turf(src))
+		else
+			if (src.flock?.getComplexDroneCount() > 1)
+				for (var/mob/living/critter/flock/drone/F in src.flock.units)
+					if (istype(F) && F != src)
+						src.controller.set_loc(get_turf(F))
+						break
```

Comment:
Why this horrible loop? Why not just send them to latejoin or blobstart whenever they exit something on another zlevel? That's what ghosts do when they're on z2 and z4

Reply:
I moved it into a proc, so only one case of it appearing in code. Other than that, I don't see the problem with it? It just makes it a little faster.

Reply:
It's faster to just not go over
Everything until you get a drone. I'd prefer `pick` if you wanted similar behaviour, or just sending them to a landmark like blob start

Reply:
Then just jump to a landmark